sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.rest;
+
+import java.util.stream.Stream;
+
+/**
+ *  This class provides utility to iterate over {@link PagedResponse}.
+ * @param  <T> value
+ */
+public class PagedIterable<T> extends IterableResponse<T>  {
+    private final PagedFlux<T> pagedFlux;
+
+    /**
+     * Creates instance given {@link PagedFlux}.
+     * @param pagedFlux to use as iterable
+     */
+    public PagedIterable(PagedFlux<T> pagedFlux) {
+        super(pagedFlux);
+        this.pagedFlux = pagedFlux;
+    }
+
+    /**
+     * Retrieve the {@link Stream}, one page at a time.
+     * It will provide same stream of T values from starting every time if called multiple times.
+     * @return {@link Stream} of {@link PagedResponse}
+     */
+    public Stream<PagedResponse<T>> streamByPage() {
+        return pagedFlux.byPage().toStream();
+    }
+
+    /**
+     * Provides iterable API for{@link PagedResponse}
```

Comment:
Iterable is a type, it should have an upper case I.
Lacking a space between 'for' and `{@link`